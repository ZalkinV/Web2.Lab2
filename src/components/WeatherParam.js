
export default function WeatherParam(props)
{
  return (
    <div>
      <div>props.name</div>
      <div>props.value</div>
    </div>
  );
}
