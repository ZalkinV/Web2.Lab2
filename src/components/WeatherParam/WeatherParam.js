import React from "react";

import "./WeatherParam.css";


export default function WeatherParam(props) {
  return (
    <div class="param">
      <div class="name">{props.name}</div>
      <div class="value">{props.value}</div>
    </div>
  );
}
