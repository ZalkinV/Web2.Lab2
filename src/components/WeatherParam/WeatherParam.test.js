import React from "react";
import renderer from "react-test-renderer";

import WeatherParam from "./WeatherParam";


describe("<WeatherParam /> render", () =>{

  const DEFAULT_PROPS = {
    name: "Temperature",
    value: "25",
  };

  test("should has name and value", () => {
    const { name, value } = DEFAULT_PROPS;
    
    const tree = renderer
      .create(
        <WeatherParam
          name={name}
          value={value}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("should has name", () => {
    const { name } = DEFAULT_PROPS;
    
    const tree = renderer
      .create(
        <WeatherParam
          name={name}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("should has value", () => {
    const { value } = DEFAULT_PROPS;
    
    const tree = renderer
      .create(
        <WeatherParam
          value={value}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("should has nothing", () => {
    const tree = renderer
      .create(
        <WeatherParam />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
