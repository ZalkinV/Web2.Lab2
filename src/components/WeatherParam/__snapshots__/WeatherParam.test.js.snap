// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherParam /> render should has name 1`] = `
<div
  className="param"
>
  <div
    className="name"
  >
    Temperature
  </div>
  <div
    className="value"
  />
</div>
`;

exports[`<WeatherParam /> render should has name and value 1`] = `
<div
  className="param"
>
  <div
    className="name"
  >
    Temperature
  </div>
  <div
    className="value"
  >
    25
  </div>
</div>
`;

exports[`<WeatherParam /> render should has nothing 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;

exports[`<WeatherParam /> render should has value 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  >
    25
  </div>
</div>
`;
